airbyte-config/config-models/src/main/java/io/airbyte/config/helpers/StateMessageHelper.java

```diff
@@ -6,26 +6,52 @@
 
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Iterables;
 import io.airbyte.commons.json.Jsons;
 import io.airbyte.config.StateType;
 import io.airbyte.config.StateWrapper;
 import io.airbyte.protocol.models.AirbyteStateMessage;
 import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
 import java.util.List;
 import java.util.Optional;
+import java.util.function.BiFunction;
 
 public class StateMessageHelper {
 
   public static class AirbyteStateMessageListTypeReference extends TypeReference<List<AirbyteStateMessage>> {}
 
+  private static final BiFunction<List<AirbyteStateMessage>, Boolean, StateWrapper> provideGlobalState = (stateMessages, useStreamCapableState) -> {
```

Comment:
What is the reason for defining this as a `BiFunction`, rather than just defining it as a normal function, e.g.
```
private static StateWrapper extractGlobalState(AirbyteStateMessage message, boolean useStreamCapableState) {
    // implementation
}
```
which feels a bit more readable.

From what I can tell this method is only used directly in this class, so I'm not seeing the benefit of using a BiFunction here.

Same comment goes for `provideStreamState`